c10/core/TensorImpl.h

```diff
@@ -230,6 +230,9 @@ struct C10_API ExtraMeta {
   SymDimVector strides_ = {1};
   SymInt numel_ = 1;
   SymInt storage_offset_ = 0;
+  bool is_contiguous_ = true;
+  bool is_channels_last_contiguous_ = false;
+  bool is_channels_last_3d_contiguous_ = false;
```

Comment:
There isn't actually any reason to store these on EmptyMeta, since they're bool (and not SymBool).

Reply:
I think in the future we might want to make them SymBool - and I think this is a bit cleaner for now?

Reply:
okey dokey